spec/controllers/application_controller/dialog_runner_spec.rb

```diff
@@ -169,6 +169,29 @@
       expect(controller).to receive(:replace_right_cell)
       controller.send(:dialog_form_button_pressed)
     end
+
+    it "redirects to show for the display/display_id if present" do
+      edit = {:rec_id     => 1,
+              :wf         => wf,
+              :key        => 'dialog_edit__10',
+              :display    => "generic_objects",
+              :display_id => 10}
+      controller.instance_variable_set(:@edit, edit)
+      controller.instance_variable_set(:@sb, {})
+      session[:edit] = edit
+
+      controller.instance_variable_set(:@_params, :button => 'submit', :id => 10)
+      allow(controller).to receive(:role_allows?).and_return(true)
```

Comment:
I believe we have some method to stub the permissions. Actually several. Not sure which one fits best here, but I suggest using one. (Actually it would be best to limit the number of different ways to stub the privileges.)